pkg/icingadb/types/notification_type.go

```diff
@@ -19,13 +20,11 @@ func (nt *NotificationType) UnmarshalText(text []byte) error {
 	if err != nil {
 		return types.CantParseUint64(err, s)
 	}
-
-	n := NotificationType(i)
-	if uint64(n) != i {
-		// Truncated due to above cast, obviously too high
+	if i > math.MaxUint16 {
```

Comment:
One line above the context shown by GitHub: `strconv.ParseUint(s, 10, 64)`

Shouldn't changing that 64 to 16 do an appropriate bounds check? :thinking: (Other question would still be if the linter was also happy with that).

Reply:
Good idea and it even worked for the linter :exploding_head: